src/tonic010.rs

```diff
@@ -31,6 +31,10 @@ impl Connected for Connection {
         if self.try_borrow_stdio().is_some() {
             return ListenerConnectInfo::Stdio;
         }
+        #[cfg(feature = "vsock")]
+        if let Some(vsock) = self.try_borrow_vsock() {
+            return ListenerConnectInfo::Vsock(vsock.connect_info())
```

Comment:
`ListenerConnectInfo::Vsock` does not currently exist (and it fails compilation), also `#[cfg]` omits platform check.

Are you sure we need to also add a special support for legacy tonic 0.10? Maybe just let it fall back to `Other` below?

Reply:
Yup, tonic010 was added by mistake. I plan to add support 0.12+ only (if you haven't mind). Otherwise it was too much copy-paste for inlining VsockConnectInfo